pkg/proxy/proxy_community_test.go

```diff
@@ -0,0 +1,374 @@
+			// Then  the request is forwarded from the proxy
+			// And   the request impersonates smith
+			// And   the request's X-SSO-User Header is set to smith's ID
+			// And   the request is NOT successful
+			"plain http actual request as non-owner to private workspace": {
+				ProxyRequestMethod:  "GET",
+				ProxyRequestHeaders: map[string][]string{"Authorization": {"Bearer " + s.token(smith)}},
+				ExpectedAPIServerRequestHeaders: map[string][]string{
+					"Authorization":    {"Bearer clusterSAToken"},
+					"Impersonate-User": {"smith"},
+					"X-SSO-User":       {"username-" + smith.String()},
+				},
+				ExpectedProxyResponseStatus: http.StatusForbidden,
+				RequestPath:                 podsRequestURL("alice-private"),
+				ExpectedResponse:            "invalid workspace request: access to workspace 'alice-private' is forbidden",
+			},
+			// Given banned user eve exists
+			// And   user smith exists
+			// And   smith owns a workspace named smith-community
+			// And   smith-community is publicly visible (shared with PublicViewer)
+			// When  eve requests the list of pods in smith's workspace
+			// Then  the proxy does NOT forward the request
+			// And   the proxy rejects the call with 403 Forbidden
+			"plain http actual request as banned user to community workspace": {
+				ProxyRequestMethod:          "GET",
+				ProxyRequestHeaders:         map[string][]string{"Authorization": {"Bearer " + s.token(eve, authsupport.WithEmailClaim(eveEmail))}},
+				ExpectedProxyResponseStatus: http.StatusForbidden,
+				RequestPath:                 podsRequestURL("smith-community"),
+				ExpectedResponse:            "user access is forbidden: user access is forbidden",
+			},
+			// Given banned user eve exists
+			// And   user alice exists
+			// And   alice owns a private workspace
+			// When  eve requests the list of pods in alice's workspace
+			// Then  the proxy does NOT forward the request
+			// And   the proxy rejects the call with 403 Forbidden
+			"plain http actual request as banned user to private workspace": {
+				ProxyRequestMethod:          "GET",
+				ProxyRequestHeaders:         map[string][]string{"Authorization": {"Bearer " + s.token(eve, authsupport.WithEmailClaim(eveEmail))}},
+				ExpectedProxyResponseStatus: http.StatusForbidden,
+				RequestPath:                 podsRequestURL("alice-private"),
+				ExpectedResponse:            "user access is forbidden: user access is forbidden",
+			},
+		}
+
+		for k, tc := range tests {
+			s.Run(k, func() {
+
+				// given
+				fakeApp.Err = nil
+
+				testServer.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("Content-Type", "application/json")
+					// Set the Access-Control-Allow-Origin header to make sure it's overridden by the proxy response modifier
+					w.Header().Set("Access-Control-Allow-Origin", "dummy")
+					w.WriteHeader(http.StatusOK)
+					_, err := w.Write([]byte("my response"))
```

Comment:
not sure I understand why do we need to set the response here , don't we already do this at lines 80-84 ? But maybe it's a different handler/server 😕

Reply:
great catch, fixed in d5c15036812a2c789cbb7e75ac22237cdf9a7159